addon/SearchProvider.js

```diff
@@ -3,7 +3,7 @@ SearchSuggestionController, PrivateBrowsingUtils, exports, require */
 
 "use strict";
 const {Ci, Cu} = require("chrome");
-const CURRENT_ENGINE = "browser-search-engine-modified";
+const SEARCH_ENGINE_TOPIC = "browser-search-engine-modified";
 const HIDDEN_ENGINES = "browser.search.hiddenOneOffs";
```

Comment:
As discussed on IRC, we'll want to watch this pref for changes. Probably just emit `SEARCH_ENGINE_TOPIC`.

https://dxr.mozilla.org/mozilla-central/source/browser/components/preferences/in-content/search.js#292-300